include/seqan3/alphabet/concept.hpp

```diff
@@ -43,219 +43,81 @@
+ * \{
  */
-template <typename alphabet_type>
-constexpr underlying_rank_t<alphabet_type> to_rank(alphabet_type const c)
-    requires requires (alphabet_type c) { { c.to_rank() } -> underlying_rank_t<alphabet_type>; }
-{
-    return c.to_rank();
-}
-
-/*!\brief Free function wrapper that calls `.assign_char(in)` on the first argument.
- * \tparam alphabet_type Must provide an `.assign_char()` member function.
- * \param c The alphabet letter that you wish to assign to.
- * \param in The `char` value you wish to assign.
- * \returns A reference to the alphabet letter you passed in.
+/*!\typedef rank_type seqan3::underlying_rank_t<semi_alphabet_concept>
+ * \brief The rank_type of the alphabet. [type metafunction / struct or templated typedef]
+ * \tparam alphabet_type Must satisfy seqan3::semi_alphabet_concept.
+ * \relates seqan3::semi_alphabet_concept
+ *
+ * This is the expression to retrieve the type (no parentheses!):
+ * ```cpp
+ * using the_rank_type = seqan3::underlying_rank_t<alphabet_type>;
+ * ```
  */
-template <typename alphabet_type>
-constexpr alphabet_type & assign_char(alphabet_type & c, underlying_char_t<alphabet_type> const in)
-    requires requires (alphabet_type c) { { c.assign_char(char{0}) } -> alphabet_type &; }
-{
-    return c.assign_char(in);
-}
-
-/*!\brief Free function wrapper that calls `.assign_char(in)` on the first argument.
- * \tparam alphabet_type Must provide an `.assign_char()` member function.
- * \param c An alphabet letter temporary.
- * \param in The `char` value you wish to assign.
- * \returns The assignment result as a temporary.
- * \details
- * Use this e.g. to newly create alphabet letters from char:
- * ~~~{.cpp}
- * auto l = assign_char(dna5{}, 'G');  // l is of type dna5
- * ~~~
+/*!\var rank_type seqan3::alphabet_size_v<semi_alphabet_concept>
+ * \brief The size of the alphabet. [value metafunction / constexpr variable template]
+ * \tparam alphabet_type Must satisfy seqan3::semi_alphabet_concept.
+ * \relates seqan3::semi_alphabet_concept
+ *
+ * This is the expression to retrieve the value (no parentheses!):
+ * ```cpp
+ * auto i = seqan3::alphabet_size_v<alphabet_type>; // will be of type seqan3::underlying_rank_t<alphabet_type>
+ * ```
  */
-template <typename alphabet_type>
-constexpr alphabet_type && assign_char(alphabet_type && c, underlying_char_t<alphabet_type> const in)
-    requires requires (alphabet_type c) { { c.assign_char(char{0}) } -> alphabet_type &; }
-{
-    return std::move(c.assign_char(in));
-}
-
-/*!\brief Free function wrapper that calls `.assign_rank(in)` on the first argument.
- * \tparam alphabet_type Must provide an `.assign_rank()` member function.
- * \param c The alphabet letter that you wish to assign to.
- * \param in The `rank` value you wish to assign.
- * \returns A reference to the alphabet letter you passed in.
+/*!\fn rank_type seqan3::to_rank(semi_alphabet_concept const a)
```

Comment:
Why is there a function documentation with no function?  😖 Where is the function and why not document it directly?